[Identity] Add logging for managed identity (#33144)
Azure/azure-sdk-for-js: sdk/identity/identity/CHANGELOG.md

## 4.7.1 (Unreleased)

### Features Added

> - `ManagedIdentityCredential` will now log the configured user-assigned identity if one is set. [#33144](https://github.com/Azure/azure-sdk-for-js/pull/33144)

Reviewer: ```suggestion
- `ManagedIdentityCredential` will now log the configured user-assigned managed identity ID. [#33144](https://github.com/Azure/azure-sdk-for-js/pull/33144)
```
